packages/react/src/components/cookieConsentCore/translations.js

```diff
@@ -0,0 +1,174 @@
+    if (translations[key][lang] && translations[key][lang] !== '') {
+      // Translation was found in given language, use it
+      translation = {
+        value: translations[key][lang],
+        fallback: false,
+        lang,
+        dir: getDir(lang),
+      };
+    } else if (translations[key][fallbackLang] && translations[key][fallbackLang] !== '') {
+      // Translation was not found in given language, but it was found in fallback language, use it
+      translation = {
+        value: translations[key][fallbackLang],
+        fallback: true,
+        lang: fallbackLang,
+        dir: getDir(fallbackLang),
+      };
+
+      // Show error message only if wanted language is defined in directions (originally from siteSettings.languages)
+      if (directions[lang]) {
+        // eslint-disable-next-line no-console
+        console.error(`Cookie consent: Missing translation: ${key}:${lang}, using fallback language: ${fallbackLang}`);
+      }
+    } else {
+      // Translation was not found in given language or fallback language, use first available translation
+      const firstLang = Object.keys(translations[key])[0];
+      // translation = translations[key][firstLang];
+      translation = {
+        value: translations[key][firstLang],
+        fallback: true,
+        lang: firstLang,
+        dir: getDir(firstLang),
+      };
+      // eslint-disable-next-line no-console
+      console.error(
+        `Cookie consent: Missing primary and fallback translation: ${key}:${lang}/${fallbackLang}, using first known language: ${firstLang}`,
+      );
+    }
+  } else {
+    // Translation is not a string, number or object
+    throw new Error(`Cookie consent: Invalid translation: ${key}, should be string, number or object`);
+  }
+
+  if (translation && translation.value !== null) {
+    // Convert translation value to string in case someone has used a number instead of a string
+    translation.value = translation.value.toString();
+
+    // Replace dollar strings in translation with corresponding data from parameters
+    translation.value = translation.value.replace(/\$\{.+?\}/g, (match) => {
+      const stripDollarAndParenthesis = match.replace(/(^\$\{|\}$)/g, '');
+      const parameter = index(parameters, stripDollarAndParenthesis);
+
+      // Parameters may be either string or an language object
+      if (typeof parameter === 'object') {
+        if (parameter[lang]) {
+          return parameter[lang];
+        }
+
+        // Use fallback language if translation is missing
+        if (parameter[fallbackLang]) {
+          return parameter[fallbackLang];
+        }
+
+        // Use first available language if translation is missing
+        const firstLang = Object.keys(parameter)[0];
+        if (firstLang) {
+          return parameter[firstLang];
+        }
+      }
+      return parameter;
+    });
+    return translation;
+  }
+  throw new Error(`Cookie consent: Missing translation: ${key}:${lang}`);
```

Comment:
This `Error` will never be thrown because the function will always return the translation object if it exists. The error will only be thrown if the [translation] object is undefined, which is checked at the beginning of the function.? 🤷 ping @Arkkimaagi

Reply:
Ok linting says it should return something but it's still unreachable code 😆